TransferMethod/Sources/SelectTransferMethodTypePresenter.swift

```diff
@@ -229,24 +233,47 @@ final class SelectTransferMethodTypePresenter {
             self.transferMethodConfigurationRepository
                 .getKeys(completion: self.getKeysHandler(success: { (result) in
                     self.loadCurrency(result)
-                    self.loadTransferMethodTypes(result)
+                    self.view?.showLoading()
+                    self.getFeeAndProcessingTime { (result) in
+                        self.loadTransferMethodTypes(result)
+                    }
                 }))
         }
     }
 
     private func selectCurrencyHandler() -> SelectTransferMethodTypeView.SelectItemHandler {
-        return { (currency) in
+        return { [weak self](currency) in
+            guard let strongSelf = self
+            else { return }
             if let currency = currency.value {
-                self.selectedCurrency = currency
-                self.trackCurrencyClick()
+                strongSelf.selectedCurrency = currency
+                strongSelf.trackCurrencyClick()
+            }
+            strongSelf.view?.showLoading()
+            strongSelf.getFeeAndProcessingTime { result in
+                strongSelf.loadTransferMethodTypes(result)
+                strongSelf.view?.reloadCountryCurrencyData()
             }
-            self.transferMethodConfigurationRepository.getKeys(completion: self.getKeysHandler(
-                success: { (result) in
-                    self.loadTransferMethodTypes(result)
-                    self.view?.reloadCountryCurrencyData()
-                }))
         }
     }
+    
+    private func getFeeAndProcessingTime(completion: @escaping (HyperwalletTransferMethodConfigurationKey?) -> Void) {
+        transferMethodConfigurationRepository
+            .getFeeAndProcessingTime(country: selectedCountry,
+                                     currency: selectedCurrency) { [weak self] (result) in
+                guard let strongSelf = self, let view = strongSelf.view else {
+                    return
+                }
+                view.hideLoading()
+                switch result {
+                case .failure(let error):
+                    view.showError(error, pageName: strongSelf.pageName, pageGroup: strongSelf.pageGroup, nil)
```

Comment:
there should be retry as well for connection error just like `strongSelf.view?.showError(error,
                                           pageName: strongSelf.pageName,
                                           pageGroup: strongSelf.pageGroup) {
                    strongSelf.loadTransferMethodKeys()
                }`

Reply:
Added retry